# Session 13 

In [1]:
import numpy as np
from scipy import stats

np.random.seed(42)

## Task 1 

In [2]:
delivery_times = [28, 32, 27, 35, 29, 31, 30, 33, 26, 34]

t_stat, p_value = stats.ttest_1samp(delivery_times, popmean=30)

print('Sample mean:', np.mean(delivery_times))
print('t-statistic:', t_stat)
print('p-value:', p_value)

Sample mean: 30.5
t-statistic: 0.5222329678670935
p-value: 0.6141172548083939


**Conclusion:** p = 0.614 is far above α = 0.05, so we **fail to reject the null hypothesis**. This sample gives no evidence that average delivery time differs from 30 minutes — the sample mean (30.5 min) is close enough to 30 that the difference is easily explained by random sampling variation.

## Task 2


In [3]:
before_steps = np.random.normal(6500, 800, 30).round().astype(int)
after_steps = np.random.normal(7400, 900, 30).round().astype(int)

t_stat, p_value = stats.ttest_ind(before_steps, after_steps)

print('Before mean:', before_steps.mean(), '| After mean:', after_steps.mean())
print('t-statistic:', t_stat)
print('p-value:', p_value)

Before mean: 6349.4 | After mean: 7290.9
t-statistic: -4.667304132689308
p-value: 1.8475466079332855e-05


**Conclusion:** p ≈ 0.0000185, far below α = 0.05 — **reject the null hypothesis**. Average daily steps rose from ~6349 to ~7291, and this jump is statistically significant, so the sample supports the claim that step counts increased after installing the app.

## Task 3

In [4]:
before_screen = np.array([8.2, 6.0, 6.5, 6.9, 5.7, 6.5, 6.5, 4.7, 7.5, 7.1])
after_screen = np.array([7.3, 4.9, 5.0, 5.8, 4.6, 6.0, 5.0, 3.4, 6.2, 6.7])

t_stat, p_value = stats.ttest_rel(before_screen, after_screen)

print('Before mean:', before_screen.mean(), '| After mean:', after_screen.mean())
print('t-statistic:', t_stat)
print('p-value:', p_value)

Before mean: 6.56 | After mean: 5.49
t-statistic: 8.968720037445152
p-value: 8.784645749820233e-06


**Conclusion:** p ≈ 0.0000088, well below α = 0.05 — **reject the null hypothesis**. Screen time dropped for every user in this sample (mean fell from ~6.56 to ~5.49 hours), and the paired t-test confirms 'Focus Mode' significantly reduced screen time rather than the drop being random noise.

## Task 4 —

In [5]:
zomato_ratings = np.array([4.1, 4.5, 3.8, 4.3, 4.6, 3.7, 4.0, 4.4, 3.9, 4.2, 4.7, 3.6, 4.1, 4.3, 3.8])
swiggy_ratings = np.array([3.9, 3.5, 4.2, 3.6, 3.4, 4.0, 3.8, 3.3, 4.1, 3.7, 3.5, 4.3, 3.6, 3.9, 4.0])

t_stat, p_value = stats.ttest_ind(zomato_ratings, swiggy_ratings)

print('Zomato mean:', zomato_ratings.mean(), '| Swiggy mean:', swiggy_ratings.mean())
print('t-statistic:', t_stat)
print('p-value:', p_value)

Zomato mean: 4.133333333333334 | Swiggy mean: 3.786666666666667
t-statistic: 2.9754216121212798
p-value: 0.0059690203051758674


**Conclusion:** p ≈ 0.00597, below α = 0.05 — **reject the null hypothesis**. In this sample, Zomato's average rating (4.13) is significantly higher than Swiggy's (3.79) — with 15 ratings per app, the gap is large enough that it's unlikely to be random noise. (Note: with such small samples, a different draw of 15 ratings per app could easily flip this result — worth treating as suggestive rather than conclusive without more data.)

## Task 5 

In [6]:
upi_amounts = np.array([
    712.45, 588.32, 601.18, 745.67, 523.89, 690.12, 634.56, 578.23,
    801.34, 512.67, 655.89, 723.45, 598.12, 667.78, 589.34, 741.23,
    612.45, 678.90, 545.67, 702.34, 619.56, 754.23, 583.45, 629.78, 660.12
])

t_stat, p_value = stats.ttest_1samp(upi_amounts, popmean=500)

print('Sample mean: Rs.', round(upi_amounts.mean(), 2))
print('t-statistic:', t_stat)
print('p-value:', p_value)

Sample mean: Rs. 646.03
t-statistic: 9.67981196914051
p-value: 9.245219373674188e-10


**Output interpretation:** p ≈ 9.25e-10, far below α = 0.05 — **reject the null hypothesis**. The sample's average daily UPI transaction (~₹646) is significantly higher than ₹500, so this data doesn't support ₹500 as the typical transaction amount for this group.